slackroll

```diff
@@ -570,7 +587,12 @@ def extend_remote_list(local_filelist, remote_list, pkgs_url=None):
     for pkg in get_remote_pkgs(local_filelist, pkgs_url):
         # Ignore packages matching blacklist
         fullname = pkg.fullname
-        matches = [x.search(fullname) for x in bl]
+        url = pkg.url(pkgs_url)
+        matches = []
+        for (pkg_regex, url_regex) in bl:
```

Comment:
You can replace this for loop and the following if with something simpler in Python. Comment added for clarity. The line's a bit long, though.

    # If the package full name and its URL match any entry in the black list
    if any((pkg_regex.search(fullname) is not None and url_regex.search(url) is not None) for (pkg_regex, url_regex) in bl:
        continue

Reply:
And here